mindsdb/integrations/libs/api_handler_generator.py

```diff
@@ -0,0 +1,535 @@
+            url = re.sub(r"{(\w+)}", 'x', url)
+            url = url.replace('/', '_').strip('_')
+            table_name = table_name_format.format(url=url, method=endpoint.method).lower()
+            self.resources[table_name] = RestApiTable(handler, endpoint=endpoint, resource_gen=self)
+
+        return self.resources
+
+    def process_resource_types(self, schemas: dict) -> dict:
+        resource_types = {}
+        for name, schema_info in schemas.items():
+            resource_types[name] = self._convert_to_resource_type(schema_info)
+
+        return resource_types
+
+    def process_endpoints(self, paths: dict) -> List[APIEndpoint]:
+        """
+        Processes the endpoints defined in the OpenAPI specification.
+
+        Args:
+            endpoints (Dict): A dictionary containing the endpoints defined in the OpenAPI specification.
+
+        Returns:
+            Dict: A dictionary containing the processed endpoints.
+        """
+        endpoints = []
+        for path, path_info in paths.items():
+            if self.api_base is not None and not path.startswith(self.api_base):
+                continue
+
+            for http_method, method_info in path_info.items():
+                if http_method != 'get':
+                    continue
+
+                parameters = self._process_endpoint_parameters(method_info['parameters']) if 'parameters' in method_info else {}
+
+                response = self._process_endpoint_response(method_info['responses'])
+                if response['type'] is None:
+                    continue
+
+                endpoint = APIEndpoint(
+                    url=path,
+                    method=http_method,
+                    params=parameters,
+                    response=response
+                )
+
+                endpoints.append(endpoint)
+
+        return endpoints
+
+    def _process_endpoint_parameters(self, parameters: list) -> Dict[str, APIEndpointParam]:
+        """
+        Processes the parameters defined in the OpenAPI specification.
+
+        Args:
+            parameters (Dict): A dictionary containing the parameters defined in the OpenAPI specification.
+
+        Returns:
+            Dict: A dictionary containing the processed parameters.
+        """
+        endpoint_parameters = {}
+        for parameter in parameters:
+
+            type_name = self.get_resource_type(parameter['schema'])
+
+            optional = 'default' in parameter['schema']
+            endpoint_parameters[parameter['name']] = APIEndpointParam(
+                name=parameter['name'],
+                type=type_name,
+                optional=optional,
+                where=parameter['in'],
+            )
+
+        return endpoint_parameters
+
+    def _process_endpoint_response(self, responses: dict):
+        response = None
+        response_path = []  # used to find list in response
+
+        if '200' not in responses:
+            return responses, response_path
```

Comment:
Should we keep returning  dict as bellow ? 
```
{
            'type': response,
            'path': response_path,
            'view': view
}
```

Reply:
Thanks! I found it during github testing bud didn't push. 
Updated